src/main/java/badgpt/util/TaskList.java

```diff
@@ -0,0 +1,184 @@
+import badgpt.exceptions.TaskNotFoundException;
+
+import badgpt.tasks.Deadline;
+import badgpt.tasks.Event;
+import badgpt.tasks.Task;
+import badgpt.tasks.ToDo;
+
+import java.util.ArrayList;
+import java.util.List;
+
+/**
+ * Represents a todo list.
+ */
+public class TaskList {
+    private List<Task> tasks = new ArrayList<>(100);
+    private TasksUi tasksUi = new TasksUi();
+    private boolean hasChanges = false;
+
+    /**
+     * Stores a Task object in the list. The list has a max size of 100.
+     *
+     * @param task The Task object to be stored in the list.
+     * @throws ListFullException If the list already has 100 tasks.
+     */
+    public void store(Task task) throws ListFullException {
+        if (tasks.size() == 100) {
+            throw new ListFullException("Your todo list is currently full.");
+        } else {
+            tasks.add(task);
+            tasksUi.printAddTask(task, tasks.size());
+            hasChanges = true;
+        }
+    }
+
+    /**
+     * Lists out all the tasks currently in the list.
+     *
+     * @throws ListEmptyException If the list does not contain any tasks.
+     */
+    public void listTasks() throws ListEmptyException {
+        if (tasks.size() == 0) {
+            throw new ListEmptyException("Your todo list is currently empty.");
+        } else {
+            for (int i = 0; i < tasks.size(); i++) {
+                tasksUi.printTask(tasks.get(i), i + 1, tasks.size());
+            }
+        }
+    }
+
+    /**
+     * Returns the current number of tasks in the list.
+     */
+    public int getListSize() {
+        return this.tasks.size();
+    }
+
+    /**
+     * Marks the task in the specified position in the list as complete.
+     *
+     * @param taskNum The position of the task in the list.
+     * @throws TaskNotFoundException If the number entered does not exist in the list.
+     * @throws SameStatusException If the task is already complete.
+     */
+    public void mark(int taskNum) throws TaskNotFoundException, SameStatusException {
+        try {
+            if (!tasks.get(taskNum).isComplete()) {
+                tasks.get(taskNum).complete();
+                tasksUi.printMarkUnmarkOutcome(tasks.get(taskNum), 0);
+                hasChanges = true;
+            } else {
+                throw new SameStatusException("The operation cannot be completed.", tasks.get(taskNum), 0);
+            }
+        } catch (IndexOutOfBoundsException e) {
+            throw new TaskNotFoundException(e.getMessage(), tasks.size());
+        }
+    }
+
+    /**
+     * Marks the task in the specified position in the list as incomplete.
+     *
+     * @param taskNum The position of the task in the list.
+     * @throws TaskNotFoundException If the number entered does not exist in the list.
+     * @throws SameStatusException If the task is already incomplete.
+     */
+    public void unmark(int taskNum) throws TaskNotFoundException, SameStatusException {
```

Comment:
Same suggestion as for markDone
```suggestion
    public void markUndone(int taskNum) throws TaskNotFoundException, SameStatusException {
```